pkg/server/config.go

```diff
@@ -58,7 +60,21 @@ func (c Config) Complete() (*server, error) {
 		return nil, fmt.Errorf("unable to construct a client to connect to the kubelets: %v", err)
 	}
 	nodes := informer.Core().V1().Nodes()
-	scrape := scraper.NewScraper(nodes.Lister(), kubeletClient, c.ScrapeTimeout)
+	store := storage.NewStorage(c.MetricResolution)
+	manageNodeScrape := scraper.NewScraper(kubeletClient, c.ScrapeTimeout, c.MetricResolution, store)
+	nodes.Informer().AddEventHandler(cache.ResourceEventHandlerFuncs{
+		AddFunc: func(node interface{}) {
+			if err := manageNodeScrape.AddNodeScraper(node.(*corev1.Node)); err != nil {
+				klog.V(1).ErrorS(err, "", "node", klog.KObj(node.(*corev1.Node)))
+			}
+		},
+		// UpdateFunc: func(oldNode, newNode interface{}) {
+		//	 manageNodeScrape.UpdateNodeScraper(newNode.(*corev1.Node))
+		// },
```

Comment:
I don't think we need to keep this dead code. Nothing should change with the scrapers when a node is updated.